### DISCUSSION TITLE
How can I check the balance of a specific token at a specific address?

### DISCUSSION THREAD

**Discussion by 621625** [gochain/web3#265] [Q&A]
How can I check the balance of a specific token at a specific address?

E. g. how many Dai tokens are there at address 0x82340 (Polygon chain)?

**Comment by treeder** [comment]
I think something like `web3 contract call --address 0x82340 --abi erc20 --function balanceOf 0xYOURADDRESS` should work.

**Comment by ghost** [comment]
I get the error querying the balance
```shell
export WEB3_RPC_URL=https://rpc.url

docker run --rm \
  gochain/web3 \
  contract call \
  --abi erc20 \
  --address 0x335fe027F1444C128a05f245b13A594CF1FD1e03 \
  --function balanceOf 0x7d46Dd2008157901011A7598a73BDB34A2ea6E4d
```


> `ERROR: Error calling constant function: failed to unpack values from 0x: abi: cannot marshal in to go type: length insufficient 0 require 32`

**Reply by treeder** [reply]
Is `0x335fe027F1444C128a05f245b13A594CF1FD1e03` the contract address?

**Reply by ghost** [reply]
Yes, and looks like I found the issue - we should pass a `WEB3_RPC_URL` to the Docker, but it is not so clear from the error that variable is missing
```shell
WEB3_RPC_URL=https://rpc.url
TOKEN_ADDRESS=0x335fe027F1444C128a05f245b13A594CF1FD1e03
BALANCE_ADDRESS=0x7d46Dd2008157901011A7598a73BDB34A2ea6E4d

# env variable
docker run --rm \
  -e WEB3_RPC_URL=$WEB3_RPC_URL \
  gochain/web3 \
  contract call \
  --abi erc20 \
  --address $TOKEN_ADDRESS \
  --function balanceOf $BALANCE_ADDRESS

# cli option
docker run --rm \
  gochain/web3 \
  --rpc-url $WEB3_RPC_URL \
  contract call \
  --abi erc20 \
  --address $TOKEN_ADDRESS \
  --function balanceOf $BALANCE_ADDRESS
```
```
10
```

Thank you for confirming the command.

**Reply by treeder** [reply]
Ahh, makes sense. Thanks for sharing your solution.
